pilot/cmd/pilot-agent/status/server.go

```diff
@@ -482,15 +490,19 @@ func (s *Server) handleStats(w http.ResponseWriter, r *http.Request) {
 		log.Errorf("failed to write agent metrics: %v", err)
 		metrics.AgentScrapeErrors.Increment()
 	}
-	if _, err := w.Write(envoy); err != nil {
-		log.Errorf("failed to write envoy metrics: %v", err)
-		metrics.EnvoyScrapeErrors.Increment()
+	if envoy != nil {
+		if _, err := io.Copy(w, envoy); err != nil {
```

Comment:
This introduces a new problem. That is, the request for envoy stats may be interrupted because the response is written too slowly.